R/guide-axis.R

```diff
@@ -280,7 +291,33 @@ GuideAxis <- ggproto(
     )
   },
 
+  build_ticks = function(key, elements, params, position = params$opposite) {
+
+    major <- Guide$build_ticks(
+      vec_slice(key, (key$.type %||% "major") == "major"),
+      elements, params, position,
+      elements$major_length
+    )
+
+    if (!params$minor.ticks || inherits(elements$minor, "element_blank")) {
+      return(major)
+    }
+
+    elements$ticks <- elements$minor
```

Comment:
Same as the above with the new object. I think it is a code smell if it is necessary to rename/overwrite inputs and call it multiple times. Can you see a way out of this?

Reply:
Yeah `Guide$build_ticks()` can now also accept a single element as its `elements` argument, so we don't need this renaming thing.